src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -1197,7 +1201,11 @@ int D4NFilterObject::D4NFilterReadOp::iterate(const DoutPrefixProvider* dpp, int
   std::string version = source->get_object_version();
   std::string prefix;
 
-  prefix = source->get_bucket()->get_name() + "_" + version + "_" + source->get_name();
```

Comment:
instead of doing this, you need to reset the version correctly.

Reply:
if versioning is suspended, the version calculated must have automatically been set to 'null', if it isn't you need to fix that.

Reply:
Got it, I'll make sure the version is correctly being calculated in `prepare`